Sources/TuistGenerator/Generator/BuildPhaseGenerator.swift

```diff
@@ -616,6 +614,9 @@ final class BuildPhaseGenerator: BuildPhaseGenerating {
 
         refs.forEach {
             let pbxBuildFile = PBXBuildFile(file: $0, settings: ["ATTRIBUTES": ["RemoveHeadersOnCopy"]])
+            if !target.isExclusiveTo(.macOS) {
+                pbxBuildFile.applyPlatformFilters([.macos])
+            }
```

Comment:
this is a bit of a topic of discussion. the logic for whether to include a given build phase is split between the target and `BuildPhaseGenerator`. some things use accessors on the target like `canEmbedSystemExtensions` and others just guard like `generateEmbedAppClipsBuildPhase`.

So question 1 is where should that logic live.

question 2 is how should it behave,  forcing a user to specify platform filters during configuration (e.g. the XPC dependency should be defined with the macOS filter by the user) or including filters that apply to the intersection of supported platforms (e.g. if a target has a mac destination, include the XPC dependency but add the platform filter)

This was just an attempt to intelligently apply the filter instead of relying on the previous implementation of `canEmbedXPCServices`.  I could revert and change `canEmbedXPCServices` to only allow for macOS exclusive targets if we want to address this later.